source/js/renderer.js

```diff
@@ -219,6 +226,7 @@ export default class Renderer
                     this._drawTile(pageIndex, scaled, this._cache.get(source.url));
                 }
             }, this);
+            this.publish("VisibleTilesDidLoad", pageIndex, this._zoomLevel);
```

Comment:
You should instead check out the hooks that are passed into this class when it's instantiated. Then you don't need to import the diva instance here, and you get access to the publish method that's in the viewer core.

https://github.com/DDMAL/diva.js/blob/develop-diva6/source/js/viewer-core.js#L484-L506